command/var_put.go

```diff
@@ -0,0 +1,472 @@
+			return 1
+		}
+		args = args[1:]
+	default:
+		// no-op - should be KV arg
+	}
+
+	sv, err := c.makeVariable(path)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Failed to parse variable data: %s", err))
+		return 1
+	}
+
+	if len(args) > 0 {
+		data, err := parseArgsData(stdin, args)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Failed to parse K=V data: %s", err))
+			return 1
+		}
+
+		for k, v := range data {
+			vs := v.(string)
+			if vs == "" {
+				if _, ok := sv.Items[k]; ok {
+					c.Ui.Warn(fmt.Sprintf("Removed item %q", k))
+					delete(sv.Items, k)
+				} else {
+					c.Ui.Warn(fmt.Sprintf("Item %q does not exist, continuing...", k))
+				}
+				continue
+			}
+			sv.Items[k] = vs
+		}
+	}
+	// Get the HTTP client
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	if enforce {
+		sv.ModifyIndex = checkIndex
+	}
+
+	if forceOverwrite {
+		sv, _, err = client.Variables().Update(sv, nil)
+	} else {
+		sv, _, err = client.Variables().CheckedUpdate(sv, nil)
+	}
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error creating variable: %s", err))
+		return 1
+	}
+
+	verbose(fmt.Sprintf("Created variable %q with modify index %v", sv.Path, sv.ModifyIndex))
+
+	var out string
+	switch c.outFmt {
+	case "json":
+		out = sv.AsJSON()
+	case "pretty-json":
+		out = sv.AsPrettyJSON()
+	case "hcl":
+		out = renderAsHCL(sv)
+	case "go-template":
+		if out, err = renderWithGoTemplate(sv, c.tmpl); err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+	case "table":
+		// the renderSVAsUiTable func writes directly to the ui and doesn't error.
+		renderSVAsUiTable(sv, c)
+		return 0
+	default:
+		return 0
+	}
+	c.Ui.Output(out)
+	return 0
+}
+
+// makeVariable creates a variable based on whether or not there is data in
+// content and the format is set.
+func (c *VarPutCommand) makeVariable(path string) (*api.Variable, error) {
+	var err error
+	out := new(api.Variable)
+	if len(c.contents) == 0 {
+		out.Path = path
+		out.Namespace = c.Meta.namespace
+		out.Items = make(map[string]string)
+		return out, nil
+	}
+	switch c.inFmt {
+	case "json":
+		err = json.Unmarshal(c.contents, out)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshaling json: %w", err)
+		}
+	case "hcl":
+		out, err = parseVariableSpec(c.contents)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing hcl: %w", err)
+		}
+	case "":
+		fmt.Println("format flag required")
+		return nil, errors.New("format flag required")
+	default:
+		return nil, fmt.Errorf("unknown format flag value")
+	}
+
+	// Handle cases where values are provided by CLI flags that modify the
+	// the created variable. Typical of a "copy" operation, it is a convenience
+	// to reset the Create and Modify metadata to zero.
+	var resetIndex bool
+
+	// Step on the namespace in the object if one is provided by flag
+	if c.Meta.namespace != "" && c.Meta.namespace != out.Path {
```

Comment:
```suggestion
	if c.Meta.namespace != "" && c.Meta.namespace != out.Namespace {
```

Found this because it results in weird CAS errors if you pipe in a variable you get from `var get`, but _only_ when you include the `-namespace` flag unnecessarily!